Add pandoc markdown support

Utsira opened the pull request:
support for pandoc markdown using panzer wrapper:

https://github.com/msprev/panzer

panzer allows for sets of pandoc arguments to be saved as style presets
and passed to pandoc.

This is a good fit for atom-script is it saves the user having to enter
lots of arguments into the “Configure Script” dialog.

This fills an important gap in the Atom ecosystem. Although there are lots of packages that can preview pandoc output, so far there are none to build a pandoc document. See discussion thread here: https://discuss.atom.io/t/using-atom-for-academic-writing/19222/19

rgbkrk commented:
What's the "language" pack that declares itself as 'Pandoc Markdown'?

Utsira commented:
It's language pandoc-flavored-markdown:

https://atom.io/packages/language-pfm

rgbkrk commented:
:+1:

rgbkrk commented:
Care to add this to the README? May need a caveat line.

Utsira commented:
A couple of things that I'd like feedback on:

1. At the moment, ".pdf" is appended to the end of the file path. If possible, it's be nice if the existing extension could be removed before ".pdf" is added. I wasn't sure how to do this. The extension that would need to be removed would probably be .md, but it could be .markdown

2. At the moment, the output path ends with ".pdf" (the only way to make pandoc produce a pdf). If however the atom-script user wanted pandoc to output a docx or an epub or whatever, they could override the "--output" argument using "Configure script/ program arguments", but they then would have to manually re-enter the path and file name they wanted as well as the extension. Is there some clever way that the atom-script user could use the "Configure script" dialog to pass just an extension string ".epub", ".docx" etc as a variable to atom-script that is then automatically appended to the path and file name?

Utsira commented:
Readme updated

rgbkrk commented:
For (1) use [`path.basename`](https://nodejs.org/api/path.html#path_path_basename_p_ext) on `context.filepath`.

For (2), I personally think you're better off making a separate extension that uses preview mode and can write out to multiple formats.

rgbkrk commented:
I'll stick this in for now and you can update it again later.

Utsira commented:
Sorry I got behind with this. 

I updated Atom but I'm having trouble testing changes to my fork. I updated my fork from the master of atom-script, but when I launch Atom in dev mode I get:

> Some installed packages could not be loaded because they contain native modules that were compiled for an earlier version of Atom.

Then when I click the rebuild button for script 2.29.0 I get the error:

> Error: Cannot find module '~/GitHub/script/node_modules/nodegit/lifecycleScripts/install'

Utsira commented:
Was able to rebuild when I launched from the dock (ie not in dev mode). Apologies, please ignore previous message

rgbkrk commented:
Node git has been ripped from master, I'd just checkout master from this repo (likely origin/master)
